src/api/api.c

```diff
@@ -914,12 +918,18 @@ int8_t z_query_reply(const z_loaned_query_t *query, const z_loaned_keyexpr_t *ke
     } else {
         opts = *options;
     }
-    z_loaned_bytes_t *loaned_payload = z_bytes_loan_mut(payload);
-    _z_value_t value = {
-        .payload = loaned_payload ? _z_bytes_wrap(loaned_payload->start, loaned_payload->len) : _z_bytes_empty(),
-        .encoding = {.id = opts.encoding.id, .schema = opts.encoding.schema}};
+    // Set value
+    _z_value_t value = {.payload = _z_bytes_empty(), .encoding = opts.encoding};
+    if (payload != NULL) {
+        z_loaned_bytes_t *loaned_payload = z_bytes_loan_mut(payload);
+        value.payload =
+            (loaned_payload != NULL) ? _z_bytes_wrap(loaned_payload->start, loaned_payload->len) : _z_bytes_empty();
```

Comment:
Second assignment of _z_bytes_empty looks weird better to avoid it.

Reply:
And also will be good idea to have some helper method to convert owned payload to _z_bytes_t